addons/sale_management/views/sale_portal_templates.xml

```diff
@@ -4,7 +4,7 @@
     <template id="sale_order_portal_content_inherit_sale_management" name="Order Options" inherit_id="sale.sale_order_portal_content">
 
         <xpath expr="//section[@id='details']//t[@t-set='display_discount']" position="after">
-            <t t-set="display_remove" t-value="sale_order.state in ('draft', 'sent') and any(line.sale_order_option_ids for line in sale_order.order_line)"/>
+            <t t-set="display_remove" t-value="sale_order.state in ('draft', 'sent') and any(line._can_be_edited_on_portal() for line in sale_order.order_line)"/>
```

Comment:
You forgot to update the code here ;).
The bin icon wouldn't have been shown if there were lines with the same products as optional lines, but not linked to the optional lines.